lib/tbd/psi.rb

```diff
@@ -2465,6 +2484,35 @@ def processTBD(
     edge[:set] = io[:building][:psi]
   end
 
+  # 'Unhinged' subsurfaces, like Tubular Daylight Device (TDD) domes, shouldn't
+  # share edges with parent surfaces. They could be floating 300mm above parent
+  # roof surface, for instance. Add parent surface ID to unhinged edges.
+  edges.values.each do |edge|
+    next if edge.has_key?(:psi)
+    next unless edge.has_key?(:surfaces)
+    next unless edge[:surfaces].size == 1
+    id, sub = edge[:surfaces].first
+    next unless holes.has_key?(id)
+    next unless holes[id].attributes.has_key?(:unhinged)
+
+    subsurface = os_model.getSubSurfaceByName(id)
+    next if subsurface.empty?
+    subsurface = subsurface.get
+    surface = subsurface.surface
+    next if surface.empty?
+    surface = surface.get
+    nom = surface.nameString
+    next unless surfaces.has_key?(nom)
+    next unless surfaces[nom].has_key?(:conditioned)
+    next unless surfaces[nom][:conditioned]
+    edge[:surfaces][nom] = {}
+
+    psi = {}
+    psi[:jamb] = val[:jamb]
+    edge[:psi] = psi
+    edge[:set] = io[:building][:psi]
+  end
```

Comment:
Part 2 of 2 solution of adding _unhinged_ subsurfaces to parent surface (see _populateTBDdads_ elsewhere in "psi.rb")